kotlintest-runner/kotlintest-runner-jvm/src/main/kotlin/io/kotlintest/runner/jvm/TestEngine.kt

```diff
@@ -12,17 +11,26 @@ import io.kotlintest.runner.jvm.spec.SharedInstanceSpecRunner
 import io.kotlintest.runner.jvm.spec.SpecRunner
 import org.slf4j.LoggerFactory
 import java.util.concurrent.Executors
+import java.util.concurrent.ThreadFactory
 import java.util.concurrent.TimeUnit
+import java.util.concurrent.atomic.AtomicInteger
 import java.util.concurrent.atomic.AtomicReference
 import kotlin.reflect.KClass
 import kotlin.reflect.jvm.jvmName
 
+internal class NamedThreadFactory(val name: String) : ThreadFactory {
```

Comment:
As this class is shared between executors, can it be extracted to a class for itself?

Reply:
It's already a class.

Reply:
Sorry, I meant to move it to it's own file, as some code that uses it are not in the `TestEngine.kt` file (it's here, but it's also use elsewhere)

Reply:
It's usable from anywhere. It can live in it's own file I suppose but it doesn't make much difference. It's just an implementation detail really.

Reply:
I've moved it into it's own file now.